### PR TITLE
Began implementing states to deploy Django applications

### PR DESCRIPTION
**Pull request by blarghmatey**

### CODE REVIEW

**Comment by shaidar** [comment]
README needs to be updated

File: django/install.sls [hunk 1]
```diff
@@ -1,10 +1,62 @@
...
+create_deployment_target_directory:
+  file.directory:
+    - name: /opt/{{ app_name }}
+    - user: {{ django.user }}
+    - group: {{ django.group }}
+    - recurse:
+        - user
+        - group
+
+{% set app_source = django.app_source %}
```
Comment (shaidar) [inline django/install.sls:22]: What is `django.app_source`?

**Comment by shaidar** [comment]
What's the purpose for `deploy.sls` since it's blank?

File: django/post_deploy.sls [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - bin_env: {{ django.django_admin_path }}
+    - user: {{ django.user }}
+
+{% set post_install_states = salt.pillar.get('django:states:post_install', []) %}
+{% if post_install_states %}
+include:
+  {% for post_install_state in post_install_states %}
+  - {{ post_install_state }}
+  {% endfor %}
+{% endif %}
```
Comment (shaidar) [inline django/post_deploy.sls:29]: When/How is this file called/triggered?

File: django/map.jinja [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-        },
+        'automatic_migrations': False,
+        'pkgs': ['git'],
+        'pip_path': '/usr/local/bin/pip3',
+        'deploy_from_git': True,
+        'settings_module': 'django_app.settings',
+        'django_admin_path': '/usr/local/bin/django-admin',
+        'user': 'deploy',
+        'group': 'deploy',
+        'requirements_file': 'requirements.txt'
```
Comment (shaidar) [inline django/map.jinja:11]: Should there be a `service` key in here?

File: django/post_deploy.sls (hunk 2 above)
Comment (blarghmatey) [reply django/post_deploy.sls:29]: If you're referring to the included state files then they get executed automatically by virtue of being loaded into the post_deploy state file which is already being executed.

File: django/install.sls (hunk 1 above)
Comment (blarghmatey) [reply django/install.sls:22]: It is a pillar attribute that gets merged into the map.jinja datastructure.

**Comment by shaidar** [comment]
👍

**Pull request merged**